ports/mimxrt10xx/common-hal/busio/SPI.c

```diff
@@ -97,115 +60,180 @@ static void config_periph_pin(const mcu_periph_obj_t *periph) {
+
+    //             if (mcu_spi_mosi_list[k].bank_idx != mcu_spi_miso_list[j].bank_idx)
+    //                 continue;
+
+    //             self->clock = &mcu_spi_sck_list[i];
+    //             self->miso_pin = &mcu_spi_miso_list[j];
+    //             self->mosi_pin = &mcu_spi_mosi_list[k];
+
+    //             break;
+    //         }
+    //     }
+    // }
+
+    if (self->clock != NULL && (self->mosi != NULL || self->miso != NULL)) {
+        self->spi = mcu_spi_banks[self->clock->bank_idx - 1];
     } else {
-        self->spi = mcu_spi_banks[self->clock_pin->bank_idx - 1];
+        if (spi_taken) {
+            mp_raise_ValueError(translate("Hardware busy, try alternative pins"));
+        } else {
+            mp_raise_ValueError(translate("Invalid SPI pin selection"));
+        }
     }
 
-    config_periph_pin(self->mosi_pin);
-    config_periph_pin(self->miso_pin);
-    config_periph_pin(self->clock_pin);
+    config_periph_pin(self->clock);
+    if (self->mosi != NULL) {
+        config_periph_pin(self->mosi);
+    }
+    if (self->miso != NULL) {
+        config_periph_pin(self->miso);
+    }
+    reserved_spi[self->clock->bank_idx - 1] = true;
 
     lpspi_master_config_t config = { 0 };
     LPSPI_MasterGetDefaultConfig(&config);
 
     // Always start at 250khz which is what SD cards need. They are sensitive to
     // SPI bus noise before they are put into SPI mode.
     config.baudRate = 250000;
-
     LPSPI_MasterInit(self->spi, &config, LPSPI_MASTER_CLK_FREQ);
-
     LPSPI_Enable(self->spi, false);
     uint32_t tcrPrescaleValue;
     self->baudrate = LPSPI_MasterSetBaudRate(self->spi, config.baudRate, LPSPI_MASTER_CLK_FREQ, &tcrPrescaleValue);
     LPSPI_Enable(self->spi, true);
 
-    claim_pin(self->clock_pin->pin);
-
-//    if (mosi_none) {
-//        self->MOSI_pin = NO_PIN;
-//    } else {
-//        gpio_set_pin_direction(mosi->number, GPIO_DIRECTION_OUT);
-//        gpio_set_pin_pull_mode(mosi->number, GPIO_PULL_OFF);
-//        gpio_set_pin_function(mosi->number, mosi_pinmux);
-//        self->MOSI_pin = mosi->number;
-        claim_pin(self->mosi_pin->pin);
-//    }
-
-//    if (miso_none) {
-//        self->MISO_pin = NO_PIN;
-//    } else {
-//        gpio_set_pin_direction(miso->number, GPIO_DIRECTION_IN);
-//        gpio_set_pin_pull_mode(miso->number, GPIO_PULL_OFF);
-//        gpio_set_pin_function(miso->number, miso_pinmux);
-//        self->MISO_pin = miso->number;
-        claim_pin(self->miso_pin->pin);
-//    }
+    claim_pin(self->clock->pin);
+    if (self->mosi != NULL) {
+        claim_pin(self->mosi->pin);
+    }
+    if (self->miso != NULL) {
+        claim_pin(self->miso->pin);
+    }
 }
 
 void common_hal_busio_spi_never_reset(busio_spi_obj_t *self) {
-//    never_reset_sercom(self->spi_desc.dev.prvt);
-
-//    never_reset_pin_number(self->clock_pin);
-//    never_reset_pin_number(self->MOSI_pin);
-//    never_reset_pin_number(self->MISO_pin);
+    // TODO
```

Comment:
These comments I'd leave since they show what should be done here later.

Reply:
I'm doing pin resetting and claiming next, so these should be in for real by my next PR. I personally find leaving in references to SERCOM etc kind of confusing which is why I tend to remove these.